### ISSUE TITLE
[Bug] eval() used for dynamic dispatch and data parsing — arbitrary code execution confirmed

### ISSUE THREAD

**Issue by Muh76** [pykale/pykale#556]
Three uses of eval() in PyKale resolve method names, optimizer names, and annotation data from runtime strings. Arbitrary code execution was confirmed during testing — not theoretical.

Affected locations:

1. kale/pipeline/fewshot_trainer.py:114
loss, acc = eval(f"self.loss_{mode}")(feature_support, feature_query)

2. kale/pipeline/fewshot_trainer.py:149
optimizer = eval(f"torch.optim.{self.optimizer}")(self.model.parameters(), lr=self.lr)
self.optimizer is a plain string argument flowing from YACS config files.

3. kale/loaddata/video_datasets.py:79-80
if 0 <= eval(line[5]) < self.n_classes:
    data.append((line[0], eval(line[1]), eval(line[2]), eval(line[5])))
Values come from a pickled pandas DataFrame annotation file.

Confirmed exploit (location 3): running eval() on a malicious pickle value __import__('os').system('echo INJECTED_VIDEO_DATASETS') printed INJECTED_VIDEO_DATASETS to stdout with exit code 0. A real shell command executed from what should be a plain integer field.

Proof that fixes are safe — getattr() and int() produce identical results for all valid inputs:
mode=train: eval=('train_loss', 0.95) | getattr=('train_loss', 0.95) | match=True
mode=val: eval=('val_loss', 0.92) | getattr=('val_loss', 0.92) | match=True
SGD/Adam/AdamW: eval and getattr return identical optimizer types | match=True
int() results: [3, 120, 245, 7] == eval() results: [3, 120, 245, 7] | Identical: True

Suggested fixes:
Location 1: getattr(self, f"loss_{mode}")(feature_support, feature_query)
Location 2: getattr(torch.optim, self.optimizer)(self.model.parameters(), lr=self.lr)
Location 3: int(line[1]), int(line[2]), int(line[5])

Environment: PyKale 0.2.0, Python 3.11.9, Windows 11 Git Bash, PyTorch 2.6.0+cpu
